commands/charm/pager.go

```diff
@@ -0,0 +1,217 @@
+	return tea.Batch(
+		m.timerTick(),
+		func() tea.Msg {
+			<-m.ctx.Done()
+			return msgQuit{}
+		},
+	)
+}
+
+func (m *pagerModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	var (
+		cmd  tea.Cmd
+		cmds []tea.Cmd
+	)
+
+	switch msg := msg.(type) {
+	case msgQuit:
+		return m, tea.Quit
+
+	case tea.KeyMsg:
+		if k := msg.String(); k == "ctrl+c" {
+			m.cancel()
+			return m, nil
+		}
+
+	case tea.WindowSizeMsg:
+		headerHeight := lipgloss.Height(m.headerView())
+		footerHeight := lipgloss.Height(m.footerView())
+		verticalMarginHeight := headerHeight + footerHeight
+
+		if !m.ready {
+			// Since this program is using the full size of the viewport we
+			// need to wait until we've received the window dimensions before
+			// we can initialize the viewport. The initial dimensions come in
+			// quickly, though asynchronously, which is why we wait for them
+			// here.
+			m.viewport = viewport.New(msg.Width, msg.Height-verticalMarginHeight)
+			m.viewport.YPosition = headerHeight
+			m.viewport.SetContent(m.buffer.String())
+			m.ready = true
+
+			// This is only necessary for high performance rendering, which in
+			// most cases you won't need.
+			//
+			// Render the viewport one line below the header.
+			m.viewport.YPosition = headerHeight + 1
+		} else {
+			m.viewport.Width = msg.Width
+			m.viewport.Height = msg.Height - verticalMarginHeight
+		}
+	case msgUpdate:
+		m.viewport.SetContent(m.buffer.String())
+		m.viewport.GotoBottom()
+	case msgTick:
+		cmds = append(cmds, m.timerTick())
+	}
+
+	// Handle keyboard and mouse events in the viewport
+	m.viewport, cmd = m.viewport.Update(msg)
+	cmds = append(cmds, cmd)
+
+	return m, tea.Batch(cmds...)
+}
+
+func (m *pagerModel) View() string {
+	if !m.ready {
+		return "\n  Loading..."
+	}
+	return fmt.Sprintf("%s\n%s\n%s", m.headerView(), m.viewport.View(), m.footerView())
+}
+
+func (m *pagerModel) headerView() string {
+	title := m.titleStyle().Render(m.title)
+	line := strings.Repeat("─", max(0, m.viewport.Width-lipgloss.Width(title)))
+	line = lipgloss.NewStyle().Foreground(lipgloss.Color("#9B9B9B")).Render(line)
+	return lipgloss.JoinHorizontal(lipgloss.Center, title, line)
+}
+
+func (m *pagerModel) footerView() string {
+	info := m.infoStyle().Render(time.Since(m.start).Truncate(time.Second).String())
+	line := strings.Repeat("─", max(0, m.viewport.Width-lipgloss.Width(info)))
+	line = lipgloss.NewStyle().Foreground(lipgloss.Color("#9B9B9B")).Render(line)
+	return lipgloss.JoinHorizontal(lipgloss.Center, line, info)
+}
+
+func (m *pagerModel) titleStyle() lipgloss.Style {
+	b := lipgloss.RoundedBorder()
+	b.Right = "├"
+	return lipgloss.NewStyle().
+		Foreground(lipgloss.Color("#dddddd")).
+		BorderStyle(b).
+		BorderForeground(lipgloss.Color("#9B9B9B")).
+		Padding(0, 1)
+}
+
+func (m *pagerModel) infoStyle() lipgloss.Style {
+	b := lipgloss.RoundedBorder()
+	b.Left = "┤"
+	return m.titleStyle().Copy().BorderStyle(b).Foreground(lipgloss.Color("#dddddd"))
```

Comment:
Is it worth adding this and 9b9b9b as color constants?

Reply:
I meant to do that 🤦🏼‍♂️  will address in the next PR